src/sentry/tasks/commits.py

```diff
@@ -49,11 +49,11 @@ def fetch_commits(release_id, user_id, refs, prev_release_id=None, **kwargs):
             start_sha = ref['previousCommit']
         elif prev_release:
             try:
-                start_sha = Commit.objects.filter(
+                start_sha = ReleaseHeadCommit.objects.filter(
                     organization_id=release.organization_id,
-                    releaseheadcommit__release=prev_release,
+                    release=prev_release,
                     repository_id=repo.id,
-                ).values_list('key', flat=True)[0]
+                ).values_list('commit__key', flat=True)[0]
```

Comment:
can we make ReleaseHeadCommit work even without the commit log? Otherwise this doesn't actually improve any of the existing behavior